### PR TITLE
[Lock] 7.0 remove deprecations in Lock Component

### PR DESCRIPTION
**Pull request by fafiebig**
| Q             | A
| ------------- | ---
| Branch?       | 7.0 
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | yes 
| Tickets       | ... 
| License       | MIT
| Doc PR        | ...

### CODE REVIEW

File: UPGRADE-7.0.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ Lock
 ----
 
  * Add parameter `$isSameDatabase` to `DoctrineDbalStore::configureSchema()`
+ * Remove `gcProbablity` workaround in `MongoDbStore::__construct()` options, use `gcProbability` instead
```
Comment (xabbuh) [inline UPGRADE-7.0.md]:
```suggestion
 * Remove the `gcProbablity` (notice the typo) option, use `gcProbability` instead
```

File: src/Symfony/Component/Lock/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ CHANGELOG
 ---
 
  * Add parameter `$isSameDatabase` to `DoctrineDbalStore::configureSchema()`
+ * Remove `gcProbablity` workaround in `MongoDbStore::__construct()` options, use `gcProbability` instead
```
Comment (xabbuh) [inline src/Symfony/Component/Lock/CHANGELOG.md]:
```suggestion
 * Remove the `gcProbablity` (notice the typo) option, use `gcProbability` instead
```

**Review by nicolas-grekas — APPROVED** [review]
Thanks.
Please note that I took care of all the other remaining deprecations in #50873

**Comment by nicolas-grekas** [comment]
Thank you @fafiebig.

**Pull request merged**